public/app/features/dashboard/dashgrid/PanelStateWrapper.tsx

```diff
@@ -512,6 +513,14 @@ export class PanelStateWrapper extends PureComponent<Props, State> {
             onChangeTimeRange={this.onChangeTimeRange}
             eventBus={dashboard.events}
           />
+          <PanelPerformanceMonitor
+            isInPanelEdit={dashboard.panelInEdit?.id === panel.id}
+            panelType={plugin.meta.id}
+            panelId={panel.id}
+            panelTitle={panel.title}
+            panelOptions={panelOptions}
+            panelFieldConfig={panel.fieldConfig}
+          />
```

Comment:
Feels to me like this should just be a hook rather than component.

Reply:
It definitely makes sense for the option logs. Not sure how that would fare with the time measurements, will look into this more. Thank you!